cmake/FindNetCDF.cmake

```diff
@@ -1,162 +1,146 @@
-list (APPEND NetCDF_libs "${NETCDF_C_LIBRARIES}")
-set (NETCDF_LIBRARIES ${NetCDF_libs})
-set (NETCDF_INCLUDE_DIRS ${NetCDF_includes})
-
-if(NETCDF_C_INCLUDE_DIRS)
-  file(STRINGS "${NETCDF_C_INCLUDE_DIRS}/netcdf_meta.h" _netcdf_ver
-    REGEX "#define[ \t]+NC_VERSION_(MAJOR|MINOR|PATCH|NOTE)")
-  
-  string(REGEX REPLACE ".*NC_VERSION_MAJOR *\([0-9]*\).*" "\\1" _netcdf_version_major "${_netcdf_ver}")
-  string(REGEX REPLACE ".*NC_VERSION_MINOR *\([0-9]*\).*" "\\1" _netcdf_version_minor "${_netcdf_ver}")
-  string(REGEX REPLACE ".*NC_VERSION_PATCH *\([0-9]*\).*" "\\1" _netcdf_version_patch "${_netcdf_ver}")
-  string(REGEX REPLACE ".*NC_VERSION_NOTE *\"\([^\"]*\)\".*" "\\1" _netcdf_version_note "${_netcdf_ver}")
-  set(NETCDF_VERSION "${_netcdf_version_major}.${_netcdf_version_minor}.${_netcdf_version_patch}${_netcdf_version_note}")
-  unset(_netcdf_version_major)
-  unset(_netcdf_version_minor)
-  unset(_netcdf_version_patch)
-  unset(_netcdf_version_note)
-  unset(_netcdf_version_lines)
+# NetCDF CXX Library ==========================================================
+if("CXX" IN_LIST NetCDF_FIND_COMPONENTS)
+    find_path(NetCDF_CXX_INCLUDE_DIR
+        NAMES netcdf
+        PATHS "${NETCDF_ROOT}/include")
+    mark_as_advanced(NetCDF_CXX_INCLUDE_DIR)
+
+    find_library(NetCDF_CXX_LIBRARY
+        NAMES netcdf-cxx4 netcdf_c++4
+        PATHS "${NETCDF_ROOT}/lib"
+        HINTS "${NetCDF_CXX_INCLUDE_DIR}/../lib")
+    mark_as_advanced(NetCDF_CXX_LIBRARY)
 endif()
 
-# handle the QUIETLY and REQUIRED arguments and set NETCDF_FOUND to TRUE if
-# all listed variables are TRUE
-include (FindPackageHandleStandardArgs)
+if(NetCDF_CXX_INCLUDE_DIR AND NetCDF_CXX_LIBRARY)
+    set(NetCDF_CXX_FOUND TRUE)
+else()
+    set(NetCDF_CXX_FOUND FALSE)
+endif()
+
+# NetCDF Fortran Library ======================================================
+if("FORTRAN" IN_LIST NetCDF_FIND_COMPONENTS)
+    find_path(NetCDF_FORTRAN_INCLUDE_DIR
+        NAMES netcdf.inc netcdf.mod
+        PATHS "${NETCDF_ROOT}/include")
+    mark_as_advanced(NetCDF_FORTRAN_INCLUDE_DIR)
+
+    find_library(NetCDF_FORTRAN_LIBRARY
+        NAMES netcdff
+        PATHS "${NETCDF_ROOT}/lib"
+        HINTS "${NetCDF_FORTRAN_INCLUDE_DIR}/../lib")
+    mark_as_advanced(NetCDF_FORTRAN_LIBRARY)
+endif()
+
+if(NetCDF_FORTRAN_INCLUDE_DIR AND NetCDF_FORTRAN_LIBRARY)
+    set(NetCDF_FORTRAN_FOUND TRUE)
+else()
+    set(NetCDF_FORTRAN_FOUND FALSE)
+endif()
+
+# =============================================================================
+include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(NetCDF
-  DEFAULT_MSG NETCDF_LIBRARIES NETCDF_INCLUDE_DIRS NETCDF_HAS_INTERFACES NETCDF_VERSION)
-
-function(FindNetCDF_get_is_parallel_aware include_dir)
-  file(STRINGS "${include_dir}/netcdf_meta.h" _netcdf_lines
-    REGEX "#define[ \t]+NC_HAS_PARALLEL[ \t]")
-  string(REGEX REPLACE ".*NC_HAS_PARALLEL[ \t]*([0-1]+).*" "\\1" _netcdf_has_parallel "${_netcdf_lines}")
```

Comment:
Not sure if we want to loose parallel netcdf detection. I was planning on using parallel output for the netcdf writer.

Reply:
Parallel detection is on lines 52-60.

Reply:
https://github.com/NOAA-OWP/ngen/blob/b59d96ac5960d202cb0f9aa1a30fa3dffe9faf7d/cmake/FindNetCDF.cmake#L51-L60

Reply:
Sorry I missed git was showing your entire file as a change and I did not find the string when I searched.